Automattic/jetpack
# My Jetpack: Remove exported type from global.d.ts file

This PR removes the exported `Purchase' type from the typescript `global.d.ts` file because it causes the global type definitions within the file to not work correctly.

Types defined within a global.d.ts file are types that can be accessed from the global scope (i.e. **without using any form of import**). In the case of My Jetpack, we're using the global.d.ts file to define types for global variables on the window object, such as `window.myJetpackInitialState`, and `window.JP_CONNECTION_INITIAL_STATE`.
If an import or export is added to the global.d.ts file, this causes typescript to not globally recognize the globally defined types.

Related to: https://github.com/Automattic/jetpack/pull/35697





## Proposed changes:


* Removed the exported `Purchase` type definition from the `global.d.ts` file. Instead I added the Purchase properties inline into the `window.myJetpackInitialState` property definitions.
* Updated `data/purchases/usePurchases.ts`: Removed the import of the `Purchase` type from the `global.d.ts` file, and instead defined a new `Purchase` type from within a `data/types.ts` file, and imported it from there.

### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:







* Prior to checking out this branch, from within `trunk` branch, open file, `projects/packages/my-jetpack/_inc/components/product-cards-section/boost-card/boost-speed-score.tsx`
* View line 27 where we destructure `siteSuffix` and `latestBoostSpeedScores` from `window.myJetpackInitialState`.
* Assuming you have ESLint working correctly in your editor, notice that the variables are underlined in red, indicating that typescript does not recognize those properties on the `myJetpackInitialState` object.
* Now checkout this branch: `git checkout update/my-jetpack-global-types`
* Now view the same destructured variables (`siteSuffix` and `latestBoostSpeedScores`). There should no error or warnings now. Hover your mouse over `myJetpackInitialState` and see that typescript recognizes all the properties.
* Next open file: `projects/packages/my-jetpack/_inc/data/purchases/use-purchases.ts`.
* Verify that the `Purchase` type here is defined and imported from the `data/types.ts` file and that there are no errors or warnings.
